src/frontend/src/hooks/AuthHooks.ts

```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect } from 'react';
+import { Nullable } from 'types/types';
+import {ApiError } from './types';
+import { supabase } from 'main';
+import { Session } from '@supabase/supabase-js';
+
+export default function useAuth() {
+    const [isLoading, setLoading] = useState<boolean>(true);
+    const [session, setSession] = useState<Session | null>();
+    const [errors, _] = useState<Nullable<ApiError>>(null);
+
+    useEffect(() => {
+        setLoading(true);
+        const fetchSession = async () => {
+            supabase.auth.getSession().then(({ data: { session } }) => {
+                setSession(session);
+            });
```

Comment:
this method syntax is synchronous, which we are prepared to use anyway in this async method. Let's ensure we keep the hooks consistent and utilise the await-task-result pattern

```suggestion
            const response = await supabase.auth.getSession();
```